pinot-broker/src/main/java/org/apache/pinot/broker/broker/ZkBasicAuthAccessControlFactory.java

```diff
@@ -101,8 +101,9 @@ public boolean hasAccess(RequesterIdentity requesterIdentity, BrokerRequest brok
 
             Optional<ZkBasicAuthPrincipal> principalOpt =
               password2principal.entrySet().stream()
-                .filter(entry -> BcryptUtils.checkpw(entry.getKey(), entry.getValue().getPassword()))
-                .map(u -> u.getValue()).filter(Objects::nonNull).findFirst();
+                      .filter(entry -> BcryptUtils.checkpwWithCache(entry.getKey(), entry.getValue().getPassword(),
+                              _userCache.getUserPasswordAuthCache()))
+                      .map(u -> u.getValue()).filter(Objects::nonNull).findFirst();
```

Comment:
i dont know what's the relationship between this and the `pinot-controller/src/main/java/org/apache/pinot/controller/api/access/ZkBasicAuthAccessControlFactory.java`. please make sure both side are consistent

Reply:
That's fine , It would be clear to use ZkBasicAuthAccessFactory

Reply:
yeah please consolidate the usage into just one. this will make future maintenance easier